### DISCUSSION TITLE
Automatic test skip

### DISCUSSION THREAD

**Discussion by crcrewso** [qatrackplus/qatrackplus#583] [Ideas]
I'm wondering if there is any interest in this behavior. 

When submitting a partial test, if the `Uncheck if you don't want to advance the due date for this test list instance` box is unchecked, all empty tests should automatically skip.

**Comment by mchamberland** [comment]
I can see the appeal of the feature, but why not just add an extra checkbox that explicitly says all empty tests will be skipped when submitted? I don’t see why this behavior should be tied to the due date checkbox.

**Reply by crcrewso** [reply]
My thought was that, based on the way we do our QA, you only ever skip a large number of tests if you're doing an out of rotation portion of a test list that you want to not count towards scheduling. I'm not sure that an independent checkbox wouldn't create as many problems as it solves.

**Reply by randlet** [reply]
I'm not really in favour of adding another way to auto skip tests. I can understand your use case and reasoning but given you can already skip tests in bulk with categories I'm wary of adding more complexity around skipping for a fairly niche case.

**Reply by crcrewso** [reply]
> I'm not really in favour of adding another way to auto skip tests. I can understand your use case and reasoning but given you can already skip tests in bulk with categories I'm wary of adding more complexity around skipping for a fairly niche case.

That would solve the problem. I'm going to have to read up on that documentation. Thank you!

**Reply by nsmela** [reply]
Would it be possible to have a default option "Skip Test checked" for creating a test? We could then unskip it if we decide to use it.

**Reply by randlet** [reply]
You should be able to do that using the calculation procedure for the test you want to skip by setting the calculation procedure to:

 ```
UTILS.set_skip("your_test_macro", True)
your_test_macro = None
```
